d2compiler/compile.go

```diff
@@ -35,6 +43,14 @@ func Compile(p string, r io.Reader, opts *CompileOptions) (*d2graph.Graph, *d2ta
 	ast, err := d2parser.Parse(p, r, &d2parser.ParseOptions{
 		UTF16Pos: opts.UTF16Pos,
 	})
+
+	if os.Getenv("D2_LSP_MODE") == "1" {
+		jsonOutput, _ := json.Marshal(LspOutputData{Ast: ast, Err: err})
+		fmt.Print(string(jsonOutput))
+		os.Exit(42)
+		return nil, nil, err
+	}
```

Comment:
This looks ok but I wonder if there's a better place for it? What if we create a `d2lib.Parse` alongside the existing `d2lib.Compile` that just does the existing parsing above, and then in `d2cli.compile` we check this env flag there, and call `d2lib.Parse` instead of `d2lib.Compile` writing this json and exiting from d2cli?

https://github.com/terrastruct/d2/blob/master/d2cli/main.go#L395

Reply:
or it could even be a `d2cli.parse` outside of `d2cli.compile`, but I think it seems better to have the os.Exit / output in d2cli instead of d2compiler